### PR TITLE
[SPARK-28716][SQL] Add id to Exchange and Subquery's stringArgs method for easier identifying their reuses in query plans

### PR DESCRIPTION
**Pull request by dbaliafroozeh**
## What changes were proposed in this pull request?

Add id to Exchange and Subquery's stringArgs method for easier identifying their reuses in query plans, for example:
```
ReusedExchange d_date_sk#827, BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint))) [id=#2710] 
```
Where `2710` is the id of the reused exchange.

## How was this patch tested?

Passes existing tests

### CODE REVIEW

**Comment by hvanhovell** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by maropu** [comment]
Can you put explain output differences with/without this pr in the PR description? Also, It'd be better to put the simple query to reproduce the case.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala [hunk 1]
```diff
@@ -62,8 +62,8 @@ class DebuggingSuite extends SparkFunSuite with SharedSQLContext {
     }
 
     val output = captured.toString()
-    assert(output.contains(
-      """== BroadcastExchange HashedRelationBroadcastMode(List(input[0, bigint, false])) ==
+    assert(output.replaceAll("\\[id=#\\d+\\]", "[id=#x]").contains(
+      """== BroadcastExchange HashedRelationBroadcastMode(List(input[0, bigint, false])), [id=#x] ==
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala]: We need to always print this id? only for reuse cases?

Comment (hvanhovell) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala]: Then you need to make two passes over the tree. The first one to figure out which exchanges are reused and the second to print the tree. This seems a bit excessive for a debugging tool.

**Comment by hvanhovell** [comment]
Merging to master.

**Pull request closed without merging**